code/iaas/logic/src/main/java/io/cattle/platform/process/agent/AgentRemove.java

```diff
@@ -40,6 +42,13 @@ public HandlerResult handle(ProcessState state, ProcessInstance process) {
             }
 
             for (Object obj : objectManager.children(agent, clz)) {
+                if (obj instanceof StoragePool) {
+                    StoragePool sp = (StoragePool)obj;
+                    if (StoragePoolConstants.TYPE.equals(sp.getKind())) {
```

Comment:
Is this looking for kind=="storagePool".  So basically anything that is not docker?